### ISSUE TITLE
Navigator doesn't render on SS 4.1.0

### ISSUE THREAD

**Issue by aNickzz** [jonom/silverstripe-betternavigator#42]
The `BetterNavigatorExtension->isAPage()` function introduced in #27 doesn't return return true when `$this->owner` is a subclass of `SiteTree`.

Changing the function like so seems to work
```php
 protected function isAPage() {
     return $this->owner
-        && $this->owner->dataRecord
-        && $this->owner->dataRecord instanceof SiteTree
-        && $this->owner->dataRecord->ID > 0;
+        && $this->owner instanceof SiteTree
+        && $this->owner->ID > 0;
 }
```
However, this causes another error
>Emergency] Uncaught TypeError: Argument 1 passed to SilverStripe\CMS\Controllers\SilverStripeNavigator::__construct() must implement interface SilverStripe\ORM\CMSPreviewable, null given

Not sure what has changed between SS3 and SS4 with the functionality of extensions, but
```php
protected function isAPage() {
    echo get_class($this->owner->dataRecord);
    exit;
    //...
}
```
prints `JonoM\BetterNavigator\Extension\BetterNavigatorExtension`

**Comment by jonom** [comment]
Is that Commander Keen? Love it.

`$this->owner->dataRecord` must be resolving to null in your second example, so it's reporting the name of the class it was called in instead of the argument passed to it. `$dataRecord` is a property of `ContentController`, not SiteTree. It sounds like you might have the extension set up wrong.

The BetterNavigatorExtension needs to be applied to a Controller (e.g. ContentController) not a model (e.g. SiteTree or a subclass of it). It is automatically applied to ContentController out of the box, and all of your SiteTree subclasses should use a Controller that extends from this, so you shouldn't need to manually apply the BetterNavigatorExtension to any Page classes.

Can you let me know if this clears things up for you?

**Comment by aNickzz** [comment]
Haha! ok yes, so I've found the problem. Ages ago I added 
```php
SilverStripe\CMS\Controllers\ContentController::remove_extension('JonoM\BetterNavigator\Extension\BetterNavigatorExtension');
```
to my `_config.php` when I first started developing as it can get in the way when doing mobile testing, but I completely forgot about it. I only remembered when I saw that `var_dump(ContentController::get_extensions());` wasn't showing BetterNavigatorExtension.

Interesting note about null getting into `get_class`, I'll have to be aware of that 'feature' for the future ;)
